patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -2648,6 +2648,193 @@
++					endingDisplay = InfoDisplayLoader.ActiveDisplays();
++				else
++					endingDisplay = startingDisplay + 12;
++
++				if (startingDisplay >= 8)
++					startingDisplay += 1;
++				endingDisplay += 1;
++
++
++				Texture2D buttonTexture = UICommon.InfoDisplayPageArrowTexture.Value;
++				bool hovering = false;
++
++				GetInfoAccIconPosition(11, startX, out int X, out int Y);
++				Vector2 buttonPosition = new Vector2(X, Y + 20);
++				if ((float)mouseX >= buttonPosition.X && (float)mouseY >= buttonPosition.Y && (float)mouseX <= buttonPosition.X + (float)buttonTexture.Width && (float)mouseY <= buttonPosition.Y + (float)buttonTexture.Height && !PlayerInput.IgnoreMouseInterface) {
++					hovering = true;
++					player[myPlayer].mouseInterface = true;
++					if (mouseLeft && mouseLeftRelease) {
++						SoundEngine.PlaySound(12);
++						mouseLeftRelease = false;
++						if (InfoDisplayLoader.ActivePages() != InfoDisplayLoader.InfoDisplayPage + 1)
++							InfoDisplayLoader.InfoDisplayPage += 1;
++						else
++							InfoDisplayLoader.InfoDisplayPage = 0;
++					}
++
++					if (!mouseText) {
++						text = (InfoDisplayLoader.ActivePages() != InfoDisplayLoader.InfoDisplayPage + 1) ? "Next Page" : "To First Page";
++						mouseText = true;
++					}
++				}
++				spriteBatch.Draw(buttonTexture, buttonPosition, new Rectangle(0, 0, buttonTexture.Width, buttonTexture.Height), Color.White, 0f, default, 1f, SpriteEffects.None, 0f);
++				if (hovering)
++					spriteBatch.Draw(TextureAssets.InfoIcon[13].Value, buttonPosition - Vector2.One * 2f, null, OurFavoriteColor, 0f, default, 1f, SpriteEffects.None, 0f);
++
++				hovering = false;
++				GetInfoAccIconPosition(0, startX, out X, out Y);
++				buttonPosition = new Vector2(X, Y + 20);
++				if ((float)mouseX >= buttonPosition.X && (float)mouseY >= buttonPosition.Y && (float)mouseX <= buttonPosition.X + (float)buttonTexture.Width && (float)mouseY <= buttonPosition.Y + (float)buttonTexture.Height && !PlayerInput.IgnoreMouseInterface) {
++					hovering = true;
++					player[myPlayer].mouseInterface = true;
++					if (mouseLeft && mouseLeftRelease) {
++						SoundEngine.PlaySound(12);
++						mouseLeftRelease = false;
++						if (InfoDisplayLoader.InfoDisplayPage != 0)
++							InfoDisplayLoader.InfoDisplayPage -= 1;
++						else
++							InfoDisplayLoader.InfoDisplayPage = InfoDisplayLoader.ActivePages() - 1;
++					}
++
++					if (!mouseText) {
++						text = (InfoDisplayLoader.InfoDisplayPage != 0) ? "Previous Page" : "To Last Page";
++						mouseText = true;
++					}
++				}
++				spriteBatch.Draw(buttonTexture, buttonPosition, new Rectangle(0, 0, buttonTexture.Width, buttonTexture.Height), Color.White, 0f, default, 1f, SpriteEffects.FlipHorizontally, 0f);
++				if (hovering)
++					spriteBatch.Draw(TextureAssets.InfoIcon[13].Value, buttonPosition - Vector2.One * 2f, null, OurFavoriteColor, 0f, default, 1f, SpriteEffects.None, 0f);
++			}
```

Comment:
Possible to move this to a separate method in `Main.tML.cs`, with `out` variables for `startingDisplay` and `endingDisplay`?